### PR TITLE
feat(sequelize): add sql transactions support

### PR DESCRIPTION
**Pull request by shubhamp-sf**
## Description

- The `SequelizeCrudRepository` now includes a `beginTransaction` method that has a usage similar to that of [loopback transactions](https://loopback.io/doc/en/lb4/Using-database-transactions.html).
- It can have different isolation levels similar to loopback.

## How Has This Been Tested?

Tested with `postgresql` and `sqlite3` database.

- [x] Tested if retrieves model instance once transaction is committed.
- [x] Tested if it can rollback transaction.
- [x] Ensured that created transactions are isolated based on the isolation level passed.
- [x] Ensured local transactions. (i.e. It should not use transaction with another repository of different datasource).
- [x] Tested in a freshly created lb4 app https://github.com/shubhamp-sf/loopback4-sequelize-transaction-example

## Checklist

- [x] DCO (Developer Certificate of Origin) [signed in all commits](https://loopback.io/doc/en/contrib/code-contrib.html)
- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [x] Usage details added in readme.

### CODE REVIEW

File: extensions/sequelize/src/sequelize/sequelize.repository.base.ts [hunk 1]
```diff
@@ -686,16 +688,22 @@ export class SequelizeCrudRepository<
...
   async syncSequelizeModel(options: SyncOptions = {}) {
-    await this.dataSource.sequelize?.models[this.entityClass.modelName]
-      .sync(options)
-      .catch(console.error);
+    if (!this.dataSource.sequelize) {
+      throw new Error(
+        'Sequelize instance is not attached to the datasource yet.',
+      );
+    }
+    await this.dataSource.sequelize.authenticate();
```
Comment (samarpanB) [inline extensions/sequelize/src/sequelize/sequelize.repository.base.ts:696]: Why authenticate ?

File: extensions/sequelize/README.md [hunk 2]
```diff
@@ -128,6 +128,84 @@ contains "Art" in their title, which belongs to category "Programming":
...
+  // We commit the transaction.
+  await tx.commit();
+} catch (error) {
+  // If the execution reaches this line, an error was thrown.
+  // We rollback the transaction.
+  await tx.rollback();
+}
+```
+
+Switching from loopback defaults to sequelize transaction is as simple as
```
Comment (samarpanB) [inline extensions/sequelize/README.md:204]: can we move this example to loopback repo in a separate PR ?

Comment (shubhamp-sf) [reply extensions/sequelize/README.md:204]: Sure. In fact, I will add a more wholesome demo project in the `examples` dir here.

Can we keep this example in readme until then or should I remove it?

File: extensions/sequelize/src/sequelize/sequelize.repository.base.ts (hunk 1 above)
Comment (shubhamp-sf) [reply extensions/sequelize/src/sequelize/sequelize.repository.base.ts:696]: This is to match the loopback behaviour of checking the db connection on initialization.

Sequelize doesn't by default checks if the connection is okay, until the first query is requested.

Hence the authenticate call is placed there.

Comment (samarpanB) [reply extensions/sequelize/src/sequelize/sequelize.repository.base.ts:696]: ok. Makes sense

File: extensions/sequelize/README.md (hunk 2 above)
Comment (samarpanB) [reply extensions/sequelize/README.md:204]: keep it here but do add it into examples in a separate PR

**Pull request merged**